community/security/src/main/java/org/neo4j/server/security/auth/FileRepositorySerializer.java

```diff
@@ -77,6 +78,15 @@ public void saveRecordsToFile( FileSystemAbstraction fileSystem, File recordsFil
 
         try
         {
+            fileSystem.create( tempFile ).close();
+            try
+            {
+                fileSystem.setPermissions( tempFile, FilePermission.OWNER_READ, FilePermission.OWNER_WRITE );
```

Comment:
since this is the only usage of new functionality this can be achieved by tempFile.setRedable / tempFile.setWritable method calls.